exp/addons/internal/controllers/clusterresourceset_controller.go

```diff
@@ -65,27 +67,52 @@ type ClusterResourceSetReconciler struct {
 	WatchFilterValue string
 }
 
-func (r *ClusterResourceSetReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
-	err := ctrl.NewControllerManagedBy(mgr).
+func (r *ClusterResourceSetReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options, syncPeriod *time.Duration) error {
+	// Setup a separate cache for the metadata watches to secrets.
+	// This way the watch does not use the LabelSelector defined at the cache which
+	// would filter to secrets having the cluster label, because secrets referred
+	// by ClusterResourceSet are not specific to a single cluster.
+	partialSecretCache, err := cache.New(mgr.GetConfig(), cache.Options{
+		Scheme:     mgr.GetScheme(),
+		Mapper:     mgr.GetRESTMapper(),
+		HTTPClient: mgr.GetHTTPClient(),
+		SyncPeriod: syncPeriod,
+	})
+	if err != nil {
+		return errors.Wrapf(err, "Failed to create metadataOnlyCache")
```

Comment:
```suggestion
		return errors.Wrapf(err, "failed to create cache for metadata only Secret watches")
```
Maybe slightly better not sure (same below)